core/src/com/unciv/logic/automation/ConstructionAutomation.kt

```diff
@@ -322,4 +338,47 @@ class ConstructionAutomation(val cityConstructions: CityConstructions){
         }
     }
 
+    private fun addReligousUnit(){
+
+        var modifier = 0f
+
+        val missionary = cityInfo.getRuleset().units.values.asSequence()
+            .filter { it.canBePurchasedWithStat(cityInfo, Stat.Faith) }
+            .filter { it.name == "Missionary"}
```

Comment:
Please filter units by their uniques, not by name, to allow modded units

Reply:
fixed